[
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "<!--e57a25ab6845829454e8d69fc972939a-->\n\nThe following sections might be updated with supplementary metadata relevant to reviewers and maintainers.\n\n<!--006a51241073e994b41acfe9ec718e94-->\n### Code Coverage\nFor detailed information about the code coverage, see the [test coverage report](https://corecheck.dev/bitcoin/bitcoin/pulls/29649).\n<!--021abf342d371248e50ceaed478a90ca-->\n### Reviews\nSee [the guideline](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#code-review) for information on the review process.\n| Type | Reviewers |\n| ---- | --------- |\n| ACK | [fanquake](https://github.com/bitcoin/bitcoin/pull/29649#pullrequestreview-1936186538) |\n\nIf your review is incorrectly listed, please react with ð to this comment and the bot will ignore it on the next update.\n<!--174a7506f384e20aa4161008e828411d-->\n### Conflicts\nReviewers, this pull request conflicts with the following ones:\n\n* [#29641](https://github.com/bitcoin/bitcoin/pull/29641) (scripted-diff: Use LogInfo/LogDebug over LogPrintf/LogPrint by maflcko)\n\nIf you consider this pull request important, please also help to review the conflicting pull requests. Ideally, start with the one that should be merged first.\n",
      "created_at" : "2024-03-13T18:11:33Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29649#issuecomment-1995237418",
      "id" : 1995237418,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29649",
      "node_id" : "IC_kwDOABII58527Ogq",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1995237418/reactions"
      },
      "updated_at" : "2024-03-14T09:47:33Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1995237418",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/39886733?v=4",
         "events_url" : "https://api.github.com/users/DrahtBot/events{/privacy}",
         "followers_url" : "https://api.github.com/users/DrahtBot/followers",
         "following_url" : "https://api.github.com/users/DrahtBot/following{/other_user}",
         "gists_url" : "https://api.github.com/users/DrahtBot/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/DrahtBot",
         "id" : 39886733,
         "login" : "DrahtBot",
         "node_id" : "MDQ6VXNlcjM5ODg2NzMz",
         "organizations_url" : "https://api.github.com/users/DrahtBot/orgs",
         "received_events_url" : "https://api.github.com/users/DrahtBot/received_events",
         "repos_url" : "https://api.github.com/users/DrahtBot/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/DrahtBot/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/DrahtBot/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/DrahtBot"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "Looking at d9318a37ec09fe0b002815a7e48710e530620ae2 again it seems that it added this extra log statement. There's a similar log entry in `Proxy::Connect()`, is that one fine to keep? I suppose it is, since that's going to be far less frequent - but haven't tried.",
      "created_at" : "2024-03-18T11:12:55Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29649#issuecomment-2003632395",
      "id" : 2003632395,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29649",
      "node_id" : "IC_kwDOABII5853bQEL",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2003632395/reactions"
      },
      "updated_at" : "2024-03-18T11:13:56Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2003632395",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/10217?v=4",
         "events_url" : "https://api.github.com/users/Sjors/events{/privacy}",
         "followers_url" : "https://api.github.com/users/Sjors/followers",
         "following_url" : "https://api.github.com/users/Sjors/following{/other_user}",
         "gists_url" : "https://api.github.com/users/Sjors/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/Sjors",
         "id" : 10217,
         "login" : "Sjors",
         "node_id" : "MDQ6VXNlcjEwMjE3",
         "organizations_url" : "https://api.github.com/users/Sjors/orgs",
         "received_events_url" : "https://api.github.com/users/Sjors/received_events",
         "repos_url" : "https://api.github.com/users/Sjors/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/Sjors/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/Sjors/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/Sjors"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "> far less frequent\r\n\r\nThose logs should only print when the users local proxy is unreachable, which should be less frequent and also a bit more justified. However it is also spammy in that case. I ran with `-proxy=unix:/path/to/tor/socket` and then after a minute, shut down the Tor daemon. Do you think this should be cleaned up as well?\r\n\r\n```\r\n2024-04-15T17:22:45Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:47Z [error] Error while reading proxy response\r\n2024-04-15T17:22:47Z RemoveLocal(zai7kzajktbkqom3edw45zyxnn6e5z5t45d2m7dhmhilgpkz2l3frmad.onion:8333)\r\n2024-04-15T17:22:47Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:48Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:48Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:48Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:48Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:49Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:49Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:49Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:49Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:50Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:50Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:50Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:50Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:51Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:51Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:51Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:51Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:52Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:52Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:52Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:53Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:53Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:54Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:54Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:54Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:54Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:55Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:55Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:55Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:55Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:56Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:56Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:56Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:56Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:22:57Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:57Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:57Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:57Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:58Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:58Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n2024-04-15T17:22:58Z connect() to /Users/matthewzipkin/Desktop/tor failed: No such file or directory (2)\r\n2024-04-15T17:22:58Z Cannot connect to socket for /Users/matthewzipkin/Desktop/tor\r\n```",
      "created_at" : "2024-04-15T17:24:47Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29649#issuecomment-2057446510",
      "id" : 2057446510,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29649",
      "node_id" : "IC_kwDOABII5856oiRu",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2057446510/reactions"
      },
      "updated_at" : "2024-04-15T17:24:47Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2057446510",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/2084648?v=4",
         "events_url" : "https://api.github.com/users/pinheadmz/events{/privacy}",
         "followers_url" : "https://api.github.com/users/pinheadmz/followers",
         "following_url" : "https://api.github.com/users/pinheadmz/following{/other_user}",
         "gists_url" : "https://api.github.com/users/pinheadmz/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/pinheadmz",
         "id" : 2084648,
         "login" : "pinheadmz",
         "node_id" : "MDQ6VXNlcjIwODQ2NDg=",
         "organizations_url" : "https://api.github.com/users/pinheadmz/orgs",
         "received_events_url" : "https://api.github.com/users/pinheadmz/received_events",
         "repos_url" : "https://api.github.com/users/pinheadmz/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/pinheadmz/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/pinheadmz/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/pinheadmz"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "Although actually the unix sockets PR didnt change this behavior much. If you start v26 with a black hole like `-proxy=127.0.0.1:9999` you get similar spam:\r\n\r\n```\r\n2024-04-15T17:28:12Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:12Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:13Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:13Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:13Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:14Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:14Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:15Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:15Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:15Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:16Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:16Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:17Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:17Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:18Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:18Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:19Z connect() to 127.0.0.1:7656 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:19Z connect() to 127.0.0.1:9999 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:19Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:20Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n2024-04-15T17:28:20Z connect() to 127.0.0.1:9050 failed after wait: Connection refused (61)\r\n```",
      "created_at" : "2024-04-15T17:29:15Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29649#issuecomment-2057454163",
      "id" : 2057454163,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29649",
      "node_id" : "IC_kwDOABII5856okJT",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2057454163/reactions"
      },
      "updated_at" : "2024-04-15T17:29:15Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2057454163",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/2084648?v=4",
         "events_url" : "https://api.github.com/users/pinheadmz/events{/privacy}",
         "followers_url" : "https://api.github.com/users/pinheadmz/followers",
         "following_url" : "https://api.github.com/users/pinheadmz/following{/other_user}",
         "gists_url" : "https://api.github.com/users/pinheadmz/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/pinheadmz",
         "id" : 2084648,
         "login" : "pinheadmz",
         "node_id" : "MDQ6VXNlcjIwODQ2NDg=",
         "organizations_url" : "https://api.github.com/users/pinheadmz/orgs",
         "received_events_url" : "https://api.github.com/users/pinheadmz/received_events",
         "repos_url" : "https://api.github.com/users/pinheadmz/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/pinheadmz/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/pinheadmz/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/pinheadmz"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "> Do you think this should be cleaned up as well?\n\nLooks like that could be a good idea.",
      "created_at" : "2024-04-15T17:30:50Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29649#issuecomment-2057456834",
      "id" : 2057456834,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29649",
      "node_id" : "IC_kwDOABII5856okzC",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2057456834/reactions"
      },
      "updated_at" : "2024-04-15T17:30:50Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2057456834",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/863730?v=4",
         "events_url" : "https://api.github.com/users/fanquake/events{/privacy}",
         "followers_url" : "https://api.github.com/users/fanquake/followers",
         "following_url" : "https://api.github.com/users/fanquake/following{/other_user}",
         "gists_url" : "https://api.github.com/users/fanquake/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/fanquake",
         "id" : 863730,
         "login" : "fanquake",
         "node_id" : "MDQ6VXNlcjg2MzczMA==",
         "organizations_url" : "https://api.github.com/users/fanquake/orgs",
         "received_events_url" : "https://api.github.com/users/fanquake/received_events",
         "repos_url" : "https://api.github.com/users/fanquake/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/fanquake/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/fanquake/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/fanquake"
      }
   }
]
